src/Compilers/CSharp/Test/Semantic/Semantics/UseSiteErrorTests.cs

```diff
@@ -1933,6 +1933,42 @@ static void Main()
                 Diagnostic(ErrorCode.ERR_NoTypeDef, @"c[null, ""A""]").WithArguments("Missing", "Missing, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null"));
         }
 
+        [Fact]
+        public void OverloadResolutionWithUseSiteErrors_WithParamsArguments_ReturnsUseSiteErrors()
+        {
+            var missingSource = @"
+public class Missing { }
+";
+
+            var libSource = @"
+public class C
+{
+    public static Missing GetMissing(params int[] args) { return null; }
+    public static void SetMissing(params Missing[] args) { }
```

Comment:
Please add a unit-test for scenarios where params is is not the only parameter of the methods.